sky/skylet/providers/lambda_labs/node_provider.py

```diff
@@ -0,0 +1,193 @@
+import logging
+import os
+import time
+from threading import RLock
+
+from ray.autoscaler.node_provider import NodeProvider
+from ray.autoscaler.tags import TAG_RAY_CLUSTER_NAME
+from sky.authentication import PRIVATE_SSH_KEY_PATH
+from sky.skylet.providers.lambda_labs.lambda_utils import LambdaLabsClient, Metadata
+from sky.utils import command_runner
+
+REMOTE_TAG_PATH_PREFIX = '/home/ubuntu/.lambda-metadata'
+LOCAL_TAG_PATH_PREFIX = '~/.lambda_labs/metadata'
+
+logger = logging.getLogger(__name__)
+
+
+def _on_local():
+    return os.path.exists(os.path.expanduser(PRIVATE_SSH_KEY_PATH))
```

Comment:
I prefer to create another indicator file, such as `~/.lambda_labs/is_remote` on the remote cluster, instead of using the `PRIVATE_SSH_KEY_PATH`, as the user could easily upload the private key and break the system.

Also, local vs remote is not absolute. If the user tries to `sky launch` on the remote cluster, the remote cluster will become local. However, the current implementation cannot support that. It should be fine for now, but it would be good to add a TODO here.

Reply:
I created an indicator file as you suggested (it is `~/.lambda_labs/.is_remote`). Also added a TODO comment.